File: IntervalListIntersections.py
class Solution:
    def intervalIntersection(self, A: List[List[int]], B: List[List[int]]) -> List[List[int]]:
        la = len(A)
        
        lb = len(B)
        
        pa = 0
        
        pb = 0
        
        ret = []
        
        while pa<la and pb<lb:
            
            cr = [None,None]
            
            cr[0] = max(A[pa][0], B[pb][0])
            
            if A[pa][1]<=B[pb][1]:
                
                cr[1] = A[pa][1]
                
                pa+=1
                
            else:
                
                cr[1] = B[pb][1]
                
                pb+=1
                
            if cr[0]<=cr[1]: 
                
                ret.append(cr)
                
        return ret
            

File: IntervalListIntersections.cpp
class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& A, vector<vector<int>>& B) 
    {
        int size1 = A.size();
        int size2 = B.size();
        
        
        vector<vector<int>> result;
        
        int i = 0;
        int j = 0;
        
        while(i < size1 && j < size2)
        {
            int start = max(A[i][0], B[j][0]);
            int end = min(A[i][1], B[j][1]);
            
            if(start <= end)
            {
                vector<int> temp;
                temp.push_back(start);
                temp.push_back(end);
                
                result.emplace_back(temp);
            }
            
            if(A[i][1] > B[j][1])
            {
                j = j + 1;
            }
            else
            {
                i = i + 1;
            }
        }
        
        return result;
    }
};